### PR TITLE
Introduce the xtask building system

### PR DESCRIPTION
**Pull request by luojia65**
This pull request is a successor to #555, it introduces the `xtask` building system into Oreboot project. The xtask system provides an easy way to build, flash and debug Oreboot with more host platforms (like Windows). The Makefile build system is still kept for compatibility.

To use `xtask` build system, you may execute the following commands:

* cargo make
* cargo flash
* cargo gdb
* cargo asm

... to build image, flash, debug or view assembly code on Oreboot platforms.

On each command, we change directory in `src/mainboard/<vendor>/<board>` or use `--mainboard` to decide which board we need to build to, `--variant` to choose the type of target board, and use `--release` to build under release profile. Parameter `--verbose` would print verbose output message to allow us to debug xtask itself.

The xtask system currently supports sunxi/nezha platform which has not upstreamed yet. Use command `cargo run`, we run xtask to view a list of commands we may use.

Xtask system may also to expanded to run emulators within command line, this feature is reserved in current commit to allow future implementation on different emulation mainboards.

### CODE REVIEW

File: .cargo/config.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[alias]
+xtask = "run --package xtask --"
+make = "xtask make"
+asm = "xtask asm"
```
Comment (orangecms) [inline .cargo/config.toml:4]: Let's call this target `objdump` so it's clear what it does, WDYT?

Comment (luojia65) [reply .cargo/config.toml:4]: Cargo objdump is surely better than cargo asm, but this overlaps a command of widely used `cargo-binutils` tool at: https://github.com/rust-embedded/cargo-binutils/tree/master/src/bin .

Comment (orangecms) [reply .cargo/config.toml:4]: Oh interesting, I didn't know - good point :-)

Comment (orangecms) [reply .cargo/config.toml:4]: What is the difference between the objdumps? I can't see it yet.
Like, all toolchains have an objdump, be it GCC, LLVM, or the wrapper in Cargo.

Comment (orangecms) [reply .cargo/config.toml:4]: So this here just invokes `llvm-objdump` etc:
https://github.com/rust-embedded/cargo-binutils/blob/83cc6c14d40c6db463e10be3e36e1a43846d58cd/src/tool.rs#L43

Now `objdump`, for a given binary with metadata (e.g., ELF), it gives you the disassembled parts, whereas `asm`, on the other hand, may have people think of doing the forward direction, assembly.

Comment (orangecms) [reply .cargo/config.toml:4]: Ah, are you saying we cannot use `cargo objdump` as an alias because it already exists?

How do you feel about `disasm`?

Comment (orangecms) [reply .cargo/config.toml:4]: Let's take it as it is for now :)

**Review by orangecms — APPROVED** [review]
Let's take this as it is for now and work out more details over time. Thank you! 🥳

**Pull request merged**